internal/certgen/output.go

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2019 Heptio
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certgen
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// writeSecret writes out a given Secret to a file.
+func writeSecret(f *os.File, secret *corev1.Secret) error {
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+	return s.Encode(secret, f)
+}
+
+func createFile(filepath string, force bool) (f *os.File, err error) {
+
```

Comment:
This can be written a different way that closes the (arguably tiny) race between Stat and Create
```go
flags := os.O_RDWR|os.O_CREATE|os.O_TRUNC
if !force {
   flags = flags | os.O_EXCL
}
err := os.OpenFile(name, flags, 0666)
```

Reply:
TIL.